app/controllers/students_controller.rb

```diff
@@ -1,9 +1,58 @@
 class StudentsController < ApplicationController
     before_action :authenticate_user!
     before_action :set_student, only: %i[ show edit update destroy ]
-    # GET /students
+    # GET /student
     def index
-      @students = Student.where(teacher: current_user.email)
+		@students = Student.where(teacher: current_user.email)
+		@tags = Set[]
+		@emails = Set[]
+
+		@courses_taken = Hash[]
+		@semesters_taken = Hash[]
+		for student in @students do
+			@tags.add(student.tags)
+
+			# Figure out each student's course/semester they have taken
+			@courses_taken[student.course_id] = Course.find(student.course_id).course_name
+			@semesters_taken[student.course_id] = Course.find(student.course_id).semester
+
+		end
+		@semesters = Set[]
+		@sections = Set[]
+		@course_names = Set[]
+        @course_ids = Array[]
+		for record in Course.all do
+			@semesters.add(record.semester)
+			@sections.add(record.section)
+			@course_names.add(record.course_name)
+            @course_ids.append(record.id)
+		end
+
+        if params[:selected_course].nil? == false or params[:selected_semester].nil? == false or params[:selected_tag].nil? == false 
```

Comment:
are we not filtering by section?

Reply:
Student page didn't have a sections column to begin with so I just assumed we aren't filtering by it.

Reply:
Sounds good